internal/junit/reporter.go

```diff
@@ -25,6 +27,99 @@ func (r *Reporter) Add(t report.TestResult) {
 	r.TestResults = append(r.TestResults, t)
 }
 
+func parseJunitFiles(junits []report.Artifact) ([]TestSuites, error) {
+	var parsed []TestSuites
+	var errs []error
+	for _, ju := range junits {
+		if ju.Error != nil {
+			errs = append(errs, fmt.Errorf("failed to retrieve junit file: %w", ju.Error))
+			continue
+		}
+		ts, err := Parse(ju.Body)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to parse junit file: %w", err))
+			continue
+		}
+		parsed = append(parsed, ts)
+	}
+	if len(errs) > 0 {
+		return parsed, fmt.Errorf("%d errors occured while evaluating junit files: %w", len(errs), errors.Join(errs...))
+	}
+	return parsed, nil
+}
+
+// reduceSuite updates "old" with values from "new".
+func reduceSuite(old TestSuite, new TestSuite) TestSuite {
+	testMap := map[string]int{}
+	for idx, tc := range old.TestCases {
+		key := fmt.Sprintf(`%s.%s`, tc.ClassName, tc.Name)
+		testMap[key] = idx
+	}
+
+	for _, tc := range new.TestCases {
+		key := fmt.Sprintf(`%s.%s`, tc.ClassName, tc.Name)
+		var idx int
+		var ok bool
+		if idx, ok = testMap[key]; !ok {
+			log.Warn().Str("test", key).Msg("Test has been found only in retries")
```

Comment:
Can't quite put my finger on it, but this message feels weird to me 😅 

1. It talks about retries, but this method (`reduceSuite`) has no knowledge of retries.
2. User won't know what this means (ignore? contact sauce? 🔮 ?).

```suggestion
			log.Warn().Str("test", key).Msg("Sanity check failed when merging related junit test suites. New test encountered without prior history.")
```

Reply:
Thanks !